users/arxiv/users/legacy/util.py

```diff
@@ -0,0 +1,146 @@
+"""Helpers and Flask application integration."""
+
+from typing import Generator, Tuple, List
+from datetime import datetime
+from contextlib import contextmanager
+import secrets
+from base64 import b64encode, b64decode
+import hashlib
+from collections import Counter
+
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.orm.session import Session
+
+from arxiv.base.globals import get_application_config, get_application_global
+
+from ..auth import scopes
+from .. import domain
+from .models import Base, DBUser, DBPolicyClass, DBEndorsement, DBSession
+from .exceptions import UnknownSession, PasswordAuthenticationFailed
+
+# TODO: timezone!
```

Comment:
👍 I couldn't figure out why `tapir_users.joined_date` was being set to a time in UTC-9

Reply:
Ok, fixed; I also added support for session expiration, per @eawoods